### PR TITLE
Config with ssm

### PR DESCRIPTION
**Pull request by SethHubbell**
The purpose of this PR is to clean up and abstract setting up configuration details for vger, such that fewer details need to be stored or updated directly in the code base. Because there are already several AWS Dependencies, I opted to leverage AWS SSM as a means by which to store both secure, and non-secure configuration details (as AWS SSM allows for parameter encryption) to not burden either Bazaarvoice dependency nor burden any new implementer with from another arbitrary dependency, but only with something that is reasonable to expect any new implementer to have along side other AWS dependencies. 

I am also working on a suite of tests, both unit and local lambda tests that will be added in a subsequent PR.

### CODE REVIEW

File: source/git_etl/serverless.yml [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
-    ENV: ${file(./env/env.${self:provider.stage}.yml):ENV}
+    #    why do we need to provide extra credentials here?
+    DATABASE_NAME: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/database_name}
+    REDSHIFT_PORT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/port}
+    AWS_RS_USER: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/username~true}
+    AWS_RS_PASS: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/password~true}
+    CLUSTER_ENDPOINT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/cluster_endpoint}
+    GIT_API_USER: ${ssm:/${opt:ssm-store}/git/${opt:stage}/api_user~true}
+    GIT_API_KEY: ${ssm:/${opt:ssm-store}/git/${opt:stage}/api_key~true}
+    ENV: ${opt:ssm-store}
```
Comment (SethHubbell) [inline source/git_etl/serverless.yml]: This is incorrectly set as `${opt:ssm-store}` and should instead be `${opt:stage}`

File: source/jira_etl/serverless.yml [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+    #    why do we need to provide extra credentials here?
+    DATABASE_NAME: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/database_name}
+    REDSHIFT_PORT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/port}
+    AWS_RS_USER: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/username~true}
+    AWS_RS_PASS: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/password~true}
+    CLUSTER_ENDPOINT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/cluster_endpoint}
+    JH_JIRAURL: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/host_url}
+    JH_USER: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/username~true}
+    JH_PASS: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/password~true}
+    ENV: ${opt:ssm-store}
```
Comment (SethHubbell) [inline source/jira_etl/serverless.yml]: This is incorrectly set as `${opt:ssm-store}` and should instead be `${opt:stage}`

File: source/web_api/serverless.yml [hunk 3]
```diff
@@ -4,21 +4,21 @@ provider:
...
+    REDSHIFT_PORT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/port}
+    AWS_RS_USER: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/username~true}
+    AWS_RS_PASS: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/password~true}
+    CLUSTER_ENDPOINT: ${ssm:/${opt:ssm-store}/redshift/${opt:stage}/cluster_endpoint}
+    JH_JIRAURL: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/host_url}
+    JH_USER: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/username~true}
+    JH_PASS: ${ssm:/${opt:ssm-store}/jira/${opt:stage}/password~true}
+    GIT_API_USER: ${ssm:/${opt:ssm-store}/git/${opt:stage}/api_user~true}
+    GIT_API_KEY: ${ssm:/${opt:ssm-store}/git/${opt:stage}/api_key~true}
+    ENV: ${opt:ssm-store}
```
Comment (SethHubbell) [inline source/web_api/serverless.yml]: This is incorrectly set as `${opt:ssm-store}` and should instead be `${opt:stage}`

File: doc/developer-guide.md [hunk 4]
```diff
@@ -57,51 +57,70 @@
...
 1. Make sure you have installed `Node.js` and install `serverless` via `npm`.
+    * Serverless needs to be version 1.22.0 or higher to work with AWS SSM
+    * `npm install -g serverless@1.22.0`
 2. Install [Docker]((https://docs.docker.com/docker-for-mac/install/#install-and-run-docker-for-mac)).
 
+#### Configuration Preparation
+We are using SSM Parameter store for some configuration details. These details are expected to be available from the parameter store:
+
+* `/<your_parameter_prefix>/team` This is used to tag AWS Resources
+* `/<your_parameter_prefix>/role` AWS Role for the Vger Service
```
Comment (MatthewBramer) [inline doc/developer-guide.md]: This should read: /{your_parameter_prefix}/{env}/role

**Comment by MatthewBramer** [comment]
Closing for now until we firm up the js constants.

**Pull request closed without merging**